ui/app/src/components/DependecySelection.tsx

```diff
@@ -0,0 +1,232 @@
+          ) : (
+              <div>
+                <h2 className="dependency-selection--subtitle" >
+                  Hard Dependencies
+                </h2>
+                <span> • Submission mandatory</span>
+                <ul className="dependency-selection--list">
+                  {
+                    deps && deps.hard
+                      ? (
+                        deps.hard.map((uri: string) => (
+                          <li className="dependency-selection--list--hard" key={uri}>{uri}</li>
+                        ))
+                      )
+                      : (
+                        <div></div>
+                      )
+                  }
+                </ul>
+                <h2 className="dependency-selection--subtitle">
+                  Soft Dependencies
+                </h2>
+                <span> • Submission optional</span>
+                <button className="dependency-selection--nav-btn" onClick={selectAllSoft}>
+                  Select All
+                </button>
+                <ul className="dependency-selection--list" >
+                  {
+                    deps && deps.soft
+                      ? (
+                        deps.soft.map((uri: string) => (
+                          <li key={uri}>
+                            <div className="dependency-selection--list--soft-checkbox" >
+                              <BlueCheckbox
+                                checked={!!checkedSoftDep[uri]}
+                                onChange={(e) => setCheckedSoftDep([uri], e.target.checked)}
+                                value={uri}
+                                color="primary"
+                              />
+                            </div>
+                            <div className="dependency-selection--list--soft-item" >{uri}</div>
+                          </li>
+                        ))
+                      )
+                      : (
+                        <div></div>
+                      )
+                  }
+                </ul>
+              </div>
+            )
+        }
+      </div>
+      <div className="dependency-selection--bottom-section">
+        {
+          (deps == null && !showDepBtn) ? (
+            <div className="centerCircularProgress">
+              <CenterCircularProgress /> <span className="dependency-selection--center-circular-progress-text" >Loading Dependencies, please wait...</span>
+            </div>
+          ) : (
+              showDepBtn ? (
+                <button className="dependency-selection--nav-btn dependency-selection--show-all" onClick={showAllDependencies}>
+                  Show All Dependencies
+                </button>
+              ) : (
+                  <div></div>
+                )
+            )
+        }
+        <p>
+          Changes in the selection of items to publish will require "all dependencies" to be recalculated.
+          Dependency calculation is computationally.
```

Comment:
Remove "Dependency calculation is computationally."

Reply:
Done